Update World Conquest II Strings

Wedge009 opened the pull request:
**Before** merging this I have some questions about some phrases in `data/campaigns/World_Conquest/lua/game_mechanics/wocopedia/help.lua` that I didn't know how to decipher (from a native English speaker's point of view):

Line 39:
> Carryover is 15%, comunitary and avoid negative amounts.

I don't know what 'comunitary' is supposed to mean. Communal? Also what does 'carry-over avoid negative amounts' supposed to mean? Gold will always be non-negative at the start of a scenario?

Line 45:
> Every chance will do different dice rolls

This is in the context of units' training. Every chance of what? Every time a new unit is recruited there will be a different chance of being trained?

Line 68 (partially edited):
> The World Conquest II era consists of factions that are build of pairs of mainline factions of which one has a healer available (Drakes, Rebels and Loyalists), and one does not (Orcs, Dwarves and Undead) the recruit list is also organized in pairs so that you sometimes have to recruit a different unit before you can recruit the units that you want. The available heroes, desertes, and random leaders also depend on your factions, the items you can get do not depend on the faction you choose.

I didn't understand what was meant by 'factions that are build of pairs of mainline factions'. Also... 'desertes'?

I'm guessing @gfgtdf as author (?) can help with the review. I found these while reviewing #5100 for @shikadiqueen.

shikadiqueen commented:
I'm not familiarized with the campaign myself, but from a cursory glance at the file I'm guessing that "deserters" was meant where it says "desertes".

Wedge009 commented:
Deserters was my first thought too, but it didn't make sense to me in the context of heroes and leaders (units). Are deserters part of the new mechanics in this campaign?

gfgtdf commented:
> Are deserters part of the new mechanics in this campaign?

Yes, you can find them in bonus points and they'll join your army. They are a special type of heroes basicially.

gfgtdf commented:
> I don't know what 'comunitary' is supposed to mean. Communal? 

It means the carryover is evenly split over the players, if say the players end a scenario with 30,100, and 200 then it will take the sum, devide it by 3 and then take   15% of that so in the next scenario so in out case every player will start the next scenario with (330)/3 * 0.15 = 16 extra gold.  

> Also what does 'carry-over avoid negative amounts' supposed to mean? Gold will always be non-negative at the start of a scenario?

That you cannot carry over negative amounts of gold.

> Every chance will do different dice rolls

think It basically means that it works as one would expect: the probabilities to get ertain training are independent of each other.

> I didn't understand what was meant by 'factions that are build of pairs of mainline factions'.

Well each factions in the WC2 era lets you recruits the units of two mainline factions, for example "The Cult" faction allows you to recruit loyalists and undead units.

stevecotton commented:
It still needs conversion of apostrophes to typographical-style ones.

soliton- commented on data/campaigns/World_Conquest/lua/game_mechanics/wocopedia/help.lua:
```diff
@@ -53,19 +53,19 @@ function wesnoth.wml_actions.wc2_show_wocopedia(cfg)
 		local str_des_feedback =
 			_ "<b>Feedback</b>:\n" ..
-			_ "For feedback plase either post in the Word conquest II thread in the official wesnoth forum https://r.wesnoth.org/t39651 or file an issue at github https://github.com/gfgtdf/World_Conquest_II/issues .\n\n" ..
+			_ "For feedback please either post in the World Conquest II thread in the official Wesnoth forum https://r.wesnoth.org/t39651 or file an issue at GitHub https://github.com/gfgtdf/World_Conquest_II/issues .\n\n" ..
```
This should presumably be rewritten for mainline if it's still useful at all.

Wedge009 replied:
Not within the scope of this PR.

soliton- commented on data/campaigns/World_Conquest/lua/game_mechanics/wocopedia/help.lua:
```diff
@@ -53,19 +53,19 @@ function wesnoth.wml_actions.wc2_show_wocopedia(cfg)
+		local str_des_items = _ "<b>Items</b>\nItems can be given to units to make them stronger. You can get artifcats in three ways: 1) By choosing an item as your starting bonus, 2) By finding it on a map in a bonus point, 3) By dropping from enemies in later scenarios. Note, however, that not all units can pick up all items."
 		local str_cat_era = _ "Factions"
-		local str_des_era = _ "<b>Factions</b>\n The Word Conquest 2 era consists of faction that are build of pairs of mainline faction of which at one has a healer available (Drakes, Rebels and Loyalists), and one does not (Orcs, Dwarves and Undead) the recruilist is also organized in pairs so that you sometimes have to recruit a different units before you can recruit the units that you want. The ava ...
+		local str_des_era = _ "<b>Factions</b>\n The World Conquest II era consists of factions that are build of pairs of mainline faction of which at one has a healer available (Drakes, Rebels and Loyalists), and one does not (Orcs, Dwarves and Undead) the recruit list is also organized in pairs so that you sometimes have to recruit a different unit before you can recruit the units that you want. The ...
```
I think the 2/II should be removed in all user-facing strings.

Wedge009 replied:
I recall the discussion in #wesnoth-dev about WC vs WC2 but my focus here is simply on updating things that seem/look nonsensical to me. Mainly spelling errors but also things that I just didn't understand as an English speaker. (As an aside, WC2 means something completely different to me altogether.)

Wedge009 commented:
Thanks everyone for the input. I've updated the PR with typographical apostrophes and revised the WoCopedia Help based on @gfgtdf's clarification. I believe it's ready for full review and (if accepted) merge now.

stevecotton commented on data/campaigns/World_Conquest/lua/game_mechanics/unittypedata.lua:
```diff
@@ -2,11 +2,11 @@ local _ = wesnoth.textdomain 'wesnoth-World_Conquest'
 	},
 	["Orcish Grunt"] = { 
-		founddialogue=_"'bout time. Been forever since I had a good fight, eh?",
+		founddialogue=_"’bout time. Been forever since I had a good fight, eh?",
```
It's not an issue for the current PR to solve, but wmlxgettext doesn't pick up any hints about context from the coding style.

Commenting on this line because, seeing only the string without context, I thought this sounded like a Fencer rather than an Orc.

stevecotton commented on data/campaigns/World_Conquest/resources/strings/effects.cfg:
```diff
@@ -82,7 +82,7 @@ _ "terrain^water" #enddef
+_ "no undead^consumed by a living being:" #enddef
 
 #define STR_UP_TO VALUE
 _ "up to"+" {VALUE}% " #enddef
```
The character between "up" and "to" is a non-breaking space, rather than a normal space.

Wedge009 replied:
Maybe that was deliberate? I don't know. Changed, as requested.

Wedge009 replied:
See @gfgtdf's remarks below.

stevecotton commented on data/campaigns/World_Conquest/lua/map/scenario_utils/plot.lua:
```diff
@@ -45,29 +45,29 @@ function add_plot(scenario, scenario_num, nplayers)
-		start_message("6", true, _ "I don't know what the lizard is babbling about, but we'll sort these 'destroyers' out quickly enough.")
+		start_message("6", true, _ "I don’t know what the lizard is babbling about, but we’ll sort these ‘destroyers’ out quickly enough.")
 
 		end_message("1,2,3", true, _ "Victory is ours! Let us take sail in search of new lands to conquer!")
```
Should "take sail" be "set sail"?

Wedge009 replied:
I assumed it was a nautical term, like 'take in the sails', but yes, probably makes more sense to be set sail.

stevecotton commented on data/campaigns/World_Conquest/lua/map/scenario_utils/plot.lua:
```diff
@@ -45,29 +45,29 @@ function add_plot(scenario, scenario_num, nplayers)
 		end_message("1,2,3", true, _ "Do not get hasty! To reach the great continents of the east we must subdue the savage coast of Moragdu, notorious for its corsairs, before we can advance further.")
 	elseif scenario_num == 3 then
-		start_message("4", true, _ "What's this? It is a long time since strangers have set foot upon Moragdu--and in such great numbers too!")
+		start_message("4", true, _ "What’s this? It is a long time since strangers have set foot upon Moragdu--and in such great numbers too!")
```
That double-hyphen should change. IIUC, the replacement should be a spaced em dash.

Wedge009 replied:
Wasn't focused on punctuation as much as spelling, plus everyone seems to follow different conventions. My preference is for spaced en dash while others use non-spaced em dash, but I'll change per your suggestion.

stevecotton commented on data/campaigns/World_Conquest/lua/map/scenario_utils/plot.lua:
```diff
@@ -45,29 +45,29 @@ function add_plot(scenario, scenario_num, nplayers)
+		start_message("2,3", true, _ "You are not the only ones with allies! We’ll push on till the end!")
 
 		end_message("1,2,3", true, _ "Onwards to the ships!")
 		end_message("1,2,3", false, _ "We have travelled great oceans and brought low mighty empires. Should we not wait until regain our forces?")
```
Should we not wait until *we* regain our forces?

I think adding the missing word helps, but it still doesn't sound right. Maybe "until our forces regain their strength".

Wedge009 replied:
Thanks for highlighting this, I didn't read everything super-thoroughly. Agreed on your rewording.

stevecotton reviewed (commented):
In the Git commit message, please leave a blank line between the first and third lines, otherwise some Git programs concatenate all the lines of the first paragraph together (and that's way too long for some UIs).

World_Conquest/lua/campaign/objectives.lua line 10: missing a hyphen in "rightclick".

Wedge009 commented:
Sorry, I normally do that but it was my first time using `git rebase` from the command-line.

Regarding right-click, I must have missed that though I already hyphenated it for GB locale. I notice a tendency for concatenation of words in US locale anyway.

gfgtdf commented on data/campaigns/World_Conquest/resources/strings/effects.cfg:
```diff
@@ -82,10 +82,10 @@ _ "terrain^water" #enddef
 
 #define STR_UP_TO VALUE
-_ "up to"+" {VALUE}% " #enddef
+_ "up to"+" {VALUE}% " #enddef
```
Whats the reason for this change ?

Wedge009 replied:
@stevecotton remarked it stood out. What's the reason for the non-break?

gfgtdf replied:
To make the dialog where this text appears in looks better,  there are quite a few other places where this addons uses nonbreaking strings, and it does that intentionally. ( https://github.com/gfgtdf/World_Conquest_II/commit/6c352fb216853308b1b8c71e5870a1675ca0a889 ) .

gfgtdf replied:
Actually i'm not 100% sure where  utils/gameplay/training.cfg  still has them becasue i changed tehse strigns at some point and i migth ahve accidentatlyl dropped them here (which  which be bad because that's the main place where they are needed.)

gfgtdf commented:
I'm not exactly sure what to do about the nonbreakable string in particular maybe we should adda  comment or translators might oversee them, but lets leave that to a seperate pr/issue.

Wedge009 commented:
Sure, after all, I understand you're the author. This only started as a means to fix obvious spelling mistakes and somehow it's become a full string review...

shikadiqueen commented:
Is it not possible to use the XML escaping syntax for non-breakable space (`&#160;`) instead? Otherwise whoever next edits that code could accidentally replace it with regular 0x20 spaces.

shikadiqueen commented:
A couple of things:

* The PR doesn't fix several more instances of "Ok" used in UI code (sometimes not translated). At least one of them (WoCopedia) should use "Close" instead since the dialog does not have a "Cancel" option alongside it.
* I am not entirely sure about the replacement of "Ok" with "Okay" in game prose (as opposed to UI). So far we had made an active effort to avoid this and other modern words in prose.

Wedge009 commented:
Again, I was only approaching this from the perspective of an English speaker. In my experience (and others may disagree), 'Ok' doesn't make sense or at the very least I would consider to be non-standard. Can either have 'OK' (especially in dialogue buttons like 'OK' vs 'Cancel) or 'Okay/okay' in prose. I vaguely recall writing style guidelines but that wasn't what I had in mind when I made those changes. Nonetheless, I'll try to review this tomorrow.

Wedge009 commented:
Have replaced the 'Ok' in dialogues and prose. Hope that's satisfactory - let me know if I missed any.

Regarding 'OK' vs 'Close', is it okay to have label as 'Close' while leaving the ID as 'ok'? This is also beyond the scope of the PR so I have left it as-is for now.

shikadiqueen commented:
> Regarding 'OK' vs 'Close', is it okay to have label as 'Close' while leaving the ID as 'ok'?

Yes, and this is in fact required since the UI engine handles special knowledge about widgets with id of `ok`.

Wedge009 commented:
Looking at the text overall, it seems to take a more informal approach as compared with the normal, story-based campaigns. Nonetheless, given the 'modern' language use brought up previously I removed the use of 'guys' as the only update since the last push a few days ago.

Just to clarify, I replaced 'heya boss' with 'hey boss' not as a greeting (as seems to be used a lot in current times) but as the older meaning (at least the one I am accustomed to) of the use of 'hey' as a means to gain attention. Perhaps impolitely, especially in the context of a superior, but these are orcs and discipline isn't what they are normally know for so I think it's okay.

shikadiqueen commented:
Would it be appropriate to have @nemaara take a look as well since they handle string-heavy content in SP?

Wedge009 commented:
If she has time, sure. Just keeping in mind that, for my work here, I'm mainly concerned about correcting spelling errors and such, not rewriting the entire thing to be 'up to standard'.

Wedge009 commented:
Haven't heard anything for a few days - any objections before merging? Does it have to be approved by someone specific under the new contract?
